File: tests/fixtures/java/EnclosingMethodTest.java
class EnclosingMethodTest
{
    public String text;

    public static void main(String[] args)
    {
        new EnclosingMethodTest() {{
            this.text = "Hello World!";
            callHelloWorld();
        }};
    }

    public void callHelloWorld()
    {
        System.out.println(this.text);
    }
}

File: tests/Cases/EnclosingMethodTest.php
<?php
declare(strict_types=1);
namespace PHPJava\Tests\Cases;

use PHPJava\IO\Standard\Output;

class EnclosingMethodTest extends Base
{
    protected $fixtures = [
        'EnclosingMethodTest',
    ];

    public function testEnclosingMethod()
    {
        static::$initiatedJavaClasses['EnclosingMethodTest']
            ->getInvoker()
            ->getStatic()
            ->getMethods()
            ->call(
                'main',
                []
            );
        $result = Output::getHeapspace();
        $this->assertEquals("Hello World!\n", $result);
    }
}
